### ISSUE TITLE
Chat service V5

### ISSUE THREAD

**Issue by carmenfan** [3drepo/3drepo.io#3045]
### Description
Migrate the Chat service to v5

### Goals
- [x] Socket IO server should be launched from v5
- [x] V4 events should work as before

### Tasks
- [x] Rewrite socket IO app to v5
- [x] Deprecate v4 session manager - rewire them to use v5
- [x] Ensure events naming is correct in front end (wait on #3003)
- [x] Change the way logout event is called (to add the event to the constants file)
   - [x] Require refactoring the queue service first
   - [x] move event listening from v4 `chatEvent.js` to the chatService.
- [x] Remove V4 socket IO app
- [x] remove direct subscription to event queue and use the eventHandler
- [x] Ensure all v4 events still work as before
- [x] Add 1 v5 event to demostrate example of events should hook up
    - [x] ensure this also submits a v4 event
    - [x] split model update to container/federation update 
- [x] Refactoring
  - [x] have a model processing service and put chat queue into chat service
  - [x]  queue to handler
- [x] write tests
  - [x] refactor socket.io to handler
  - [x] add an extra test stage for drivers
  - [x] e2e test for chat service
- [x] Rewrite DM to a channel event
- [x] Refactor event subscription/chat broadcast to a separate file
- [x] move v4 compatibility back to v4 - just get them to listen to the messages
- [x] put socket header in constants
- [x] Add documentation to connect to chat service
- [x] investigate if we can just use the backend server (tried, won't fix for now)
- [x] Final test for sanity
